adminSiteServer/mockSiteRouter.ts

```diff
@@ -304,28 +304,28 @@ getPlainRouteWithROTransaction(
     "/data-insights{/:pageNumberOrSlug}",
     async (req, res, trx) => {
         const topicName = req.query.topic as string | undefined
-        const topic: TopicTag | undefined = topicName
+        const topicTag: TopicTag | undefined = topicName
             ? {
                   name: topicName,
                   slug: await getSlugForTopicTag(trx, topicName),
               }
             : undefined
 
         const totalPageCount = calculateDataInsightIndexPageCount(
-            await db.getPublishedDataInsightCount(trx, topic?.slug)
+            await db.getPublishedDataInsightCount(trx, topicTag?.slug)
         )
 
-        if (topic && topic.slug !== undefined) {
+        if (topicTag && topicTag.slug !== undefined) {
             // if topic slug is not a valid topic, return NotFound page
             const validTopicSlugs = await db.getAllTopicSlugs(trx)
-            if (!validTopicSlugs.includes(topic.slug)) {
+            if (!validTopicSlugs.includes(topicTag.slug)) {
                 return res.status(404).send(renderNotFoundPage())
```

Comment:
Tried this on staging, and it doesn't seem to work. I get unfiltered data insights instead of a 404. But is 404 even the behavior we get in prod? It should ideally match, but not a big deal.